### ISSUE TITLE
read.argo should create a consistent `flags` list

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#909]
The current version of `read.argo()` creates a `flags` list that looks like:
```r
> names(argo[['flags']])
[1] "pressureQc"            "pressureAdjustedQc"   
[3] "salinityQc"            "salinityAdjustedQc"   
[5] "temperatureQc"         "temperatureAdjustedQc"
```
where "Qc" is appended to the end of the field the flag is for. This is a result of the name of the flag field in the netCDF file, but to be consistent with the developing paradigm for flags and the recently added `handleFlags()` methods, the name of the flag field should match the name of the field in `@data` which it is matched to.

**Comment by richardsc** [comment]
I think I've fixed this in commit 14edc826b3db2d15d66eec45f289f2960af2e360 on my fork, but I'm not as close to this code as @dankelley so it should probably be tested thoroughly before I make a pull request.

The same commit also fixes the character vs numeric format for the flags in `argo` objects.

**Comment by richardsc** [comment]
Waiting for @dankelley to update `develop` from the `kelley` branch as per our phone conversation, then I'll look at merging this into develop.

Note that this will require rebuilding the `argo` dataset.

**Comment by dankelley** [comment]
I merged about an hour ago and pushed. So I guess it’s a good time to try this.

**Comment by richardsc** [comment]
This has been implemented in PR #910. I'd appreciate if @dankelley could have a look at the relevant changes, particularly those from commit 14edc826b3db2d15d66eec45f289f2960af2e360.

Note also that the `argo` dataset has been rebuilt to reflect the changes.

**Comment by dankelley** [comment]
Clark, this seems to be ready to close.
